### DISCUSSION TITLE
How to copy dynamic libraries (.dll) into binary output directory after build?

### DISCUSSION THREAD

**Discussion by xahon** [KDAB/cxx-qt#906] [Q&A]
I want to automate build of my rust project using Qt and copy all required dlls into the output directory. How do I do that? I don't want to mess around with hardcoding paths to libraries, is there a flexible cross-platform solution?

**Comment by ahayzen-kdab** [comment]
This is more a problem on the deployment side than on the building side, but in the Qt world there are tools to help with this.

Eg on Windows there is `windeployqt` that attempts to copy the relevant libraries to your runtime folder https://doc.qt.io/qt-6/windows-deployment.html#the-windows-deployment-tool  For linux there are tools like `linuxdeployqt` to make an appimage or using a flatpak with the KDE runtime would likely work. And I assume there are similar tools available for macOS.

Note that you could also use static linking with Qt instead as discussed in the Qt article https://doc.qt.io/qt-6/deployment.html

So you'll likely want to call these command line tools as part of your build, I'm unsure whether we can do something in the `build.rs`, but maybe we could extend `qt-build-utils` to provide enough info to know the paths that would need to be copied ?

**Comment by klochowicz** [comment]
if you're using CMake, perhaps this could be used for a cross-platform solution `qt_generate_deploy_app_script`  `https://doc.qt.io/qt-6/qt-generate-deploy-app-script.html` (there's also a QML-specific way).

It worked better for me than `windeployqt`.
